cg/meta/store/microsalt.py

```diff
@@ -0,0 +1,182 @@
+
+    new_analysis = add_new_analysis(bundle_data, microbial_order_obj, status, version_obj)
+    version_date = version_obj.created_at.date()
+
+    LOG.info("new bundle added: %s, version %s", bundle_obj.name, version_date)
+    hk_api.include(version_obj)
+    hk_api.add_commit(bundle_obj, version_obj)
+
+    return new_analysis
+
+
+def add_microbial_analysis(config_stream):
+    """Gather information from microSALT analysis to store."""
+
+    deliverables = ruamel.yaml.safe_load(config_stream)
+    new_bundle = build_microbial_bundle(deliverables)
+
+    return new_bundle
+
+
+def build_microbial_bundle(deliverables: dict) -> dict:
+    """Create a new bundle to store in Housekeeper"""
+
+    project_name = _get_microbial_name(deliverables)
+    files = microbial_deliverables_files(deliverables)
+
+    data = {
+        "name": str(project_name),
+        "created": _get_date_from_results_path(deliverables, project_name),
+        "files": files,
+    }
+    return data
+
+
+def microbial_deliverables_files(deliverables: dict) -> list:
+    """Get all deliverable files from the pipeline"""
+
+    pipeline_tags = HK_TAGS["microsalt"]
+    analysis_type_tags = MICROSALT_TAGS
+    files = parse_microbial_files(deliverables, pipeline_tags, analysis_type_tags)
+
+    return files
+
+
+def parse_microbial_files(
+    deliverables: dict, pipeline_tags: list, analysis_type_tags: dict
+) -> list:
+    """Get all files and their tags from the deliverables files"""
+
+    parsed_files = []
+    for file_ in deliverables["files"]:
+        tag_map_key = (file_["step"],) if file_["tag"] is None else (file_["step"], file_["tag"])
+        parsed_file = {
+            "path": file_["path"],
+            "tags": get_tags(file_, pipeline_tags, analysis_type_tags, tag_map_key),
+            "archive": False,
+            "tag_map_key": tag_map_key,
+        }
+        parsed_files.append(parsed_file)
+
+        if file_["path_index"]:
+            parsed_index_file = {
+                "path": file_["path_index"],
+                "tags": get_tags(
+                    file_, pipeline_tags, analysis_type_tags, tag_map_key, is_index=True
+                ),
+                "archive": False,
+                "tag_map_key": tag_map_key,
+            }
+            parsed_files.append(parsed_index_file)
+
+    return parsed_files
+
+
+def get_tags(
+    file: dict, pipeline_tags: list, tag_map: dict, tag_map_key: tuple, is_index: bool = False
+) -> list:
+    """Get all tags for a file"""
+
+    tags = {
+        "format": file["format"],
+        "id": file["id"],
+        "step": file["step"],
+        "tag": file["tag"],
+    }
+
+    tags["pipeline"] = pipeline_tags[0]
+    tags["application"] = pipeline_tags[1] if len(pipeline_tags) > 1 else None
+
+    tags = _convert_tags(tags, tag_map, tag_map_key, is_index)
+    return tags
+
+
+def _convert_tags(tags: dict, tag_map: dict, tag_map_key: tuple, is_index: bool = False) -> list:
```

Comment:
Would it be possible to reuse the `_convert_deliverables_tags_to_hk_tags` in cg/meta/store/base.py. I think we should not duplicates unless we expect them to diverge fast